atomic_reactor/plugins/post_pulp_pull.py

```diff
@@ -48,26 +49,34 @@ def __init__(self, tasker, workflow,
         :param retry_delay: int, seconds between pull attempts
         :param insecure: bool, allow non-https pull if true
         :param secret: str, path to secret
+        :param expect_v2schema2: bool, retry until Pulp retries schema 2 digest
         """
         # call parent constructor
         super(PulpPullPlugin, self).__init__(tasker, workflow)
         self.timeout = timeout
         self.retry_delay = retry_delay
         self.insecure = insecure
         self.secret = secret
+        self.expect_v2schema2 = expect_v2schema2
 
     def retry_if_not_found(self, func, *args, **kwargs):
         start = time()
 
         while True:
             try:
-                return func(*args, **kwargs)
+                digests = func(*args, **kwargs)
             except requests.exceptions.HTTPError as ex:
                 # Retry for 404 not-found because we assume Crane has
                 # not spotted the new Pulp content yet. For all other
                 # errors, give up.
                 if ex.response.status_code != requests.codes.not_found:
                     raise
+            else:
+                if not self.expect_v2schema2:
```

Comment:
Would it be easier to read as: `if not self.expect_v2schema2 or digests.v2:` ?

Reply:
It would, but for debugging purposes I added a warning message to identify we've encountered this bug - and now a full `if` is required

Reply:
Maybe an `elif` then.

Reply:
Right, looks better, fixed